test/postgresql.initialization.test.js

```diff
@@ -0,0 +1,39 @@
+// Copyright IBM Corp. 2015. All Rights Reserved.
+// Node module: loopback-connector-postgresql
+// This file is licensed under the Artistic License 2.0.
+// License text available at https://opensource.org/licenses/Artistic-2.0
+
+'use strict';
+
+var connector = require('..');
+var DataSource = require('loopback-datasource-juggler').DataSource;
+var postgresql = require('../lib/postgresql.js');
+var should = require('should');
+
+describe('initialization', function() {
+  it('honours user-defined pg-pool settings', function(done) {
+    var settings = {'max': 999}; // non-default value
+    var dataSource = new DataSource(connector, settings);
+    postgresql.initialize(dataSource, function() {
+      var pool = dataSource.connector.pg.pool;
```

Comment:
Since we can't verify pool state, we can make sure pg was not set before:

```
should.not.exist(dataSource.connector.pg);
```

to verify state before your pg pool tests.